src/main/java/com/clickhouse/kafka/connect/sink/kafka/RangeContainer.java

```diff
@@ -42,18 +42,29 @@ public boolean isInRange(long offset) {
         return false;
     }
 
+    /**
+     * This compares the stored state with the actual state
+     * @param rangeContainer A container with the actual state
+     * @return The state of the comparison
+     */
     public RangeState getOverLappingState(RangeContainer rangeContainer) {
+        long actualMinOffset = rangeContainer.getMinOffset();
+        long actualMaxOffset = rangeContainer.getMaxOffset();
+
         // SAME State [0,10] Actual [0,10]
-        if (maxOffset == rangeContainer.getMaxOffset() && minOffset == rangeContainer.getMinOffset())
+        if (maxOffset == actualMaxOffset && minOffset == actualMinOffset)
             return RangeState.SAME;
         // NEW State [0,10] Actual [11,20]
-        if (maxOffset < rangeContainer.minOffset)
+        if (actualMinOffset > maxOffset)
```

Comment:
We should probably add a diagram with the different states that you created earlier

Reply:
https://github.com/ClickHouse/clickhouse-kafka-connect/blob/main/docs/DESIGN.md 
we just need to add to current system